app/src/main/java/com/gathering/android/event/eventdetail/EventDetailViewModel2.kt

```diff
@@ -0,0 +1,214 @@
+package com.gathering.android.event.eventdetail
+
+import android.icu.text.SimpleDateFormat
+import android.location.Geocoder
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.gathering.android.R
+import com.gathering.android.auth.model.User
+import com.gathering.android.common.RESPONSE_IS_NOT_SUCCESSFUL
+import com.gathering.android.common.ResponseState
+import com.gathering.android.common.UserRepo
+import com.gathering.android.common.getDay
+import com.gathering.android.common.getHour
+import com.gathering.android.common.getMinute
+import com.gathering.android.common.getMonth
+import com.gathering.android.common.getYear
+import com.gathering.android.event.Event
+import com.gathering.android.event.eventdetail.acceptrepo.AttendanceStateRepo
+import com.gathering.android.event.model.Attendee
+import com.gathering.android.event.model.EventLocation
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.withContext
+import java.util.Calendar
+import java.util.Date
+import java.util.Locale
+import javax.inject.Inject
+
+class EventDetailViewModel2 @Inject constructor(
```

Comment:
The name of this class should be `EventDetailViewModel` You can remove `2` from the end of it.